src/cpu/instr.rs

```diff
@@ -40,30 +94,230 @@ struct Operation {
+            },
+            repr: |enc| format!("mod r{0}, r{1}, r{2}", enc.rd, enc.rs1, enc.rs2),
+        });
+        m.insert(Opcode::CMP as u8, Operation {
+            execute: |enc, cpu| {
+                let _out: u32 = cpu.state.reg[enc.rs1 as usize] as u32 - cpu.state.reg[enc.rs2 as usize] as u32;
+                cpu.state.flags = gen_flag(true, cpu.state.reg[enc.rs1 as usize] as u32, cpu.state.reg[enc.rs1 as usize] as u32, _out)
+            },
+            repr: |enc| format!("cmp r{}, r{}", enc.rs1, enc.rs2),
+        });
+        m.insert(Opcode::CMI as u8, Operation {
+            execute: |enc, cpu| {
+                let _out: u32 = cpu.state.reg[enc.rs1 as usize] as u32 - enc.imm as u32;
+                cpu.state.flags = gen_flag(true, cpu.state.reg[enc.rs1 as usize] as u32, enc.imm as u32, _out);
+            },
+            repr: |enc| format!("cmp r{}, {}", enc.rs1, enc.imm),
+        });
+        m.insert(Opcode::JAL as u8, Operation {
+            execute: |enc, cpu| {
+                cpu.state.reg[enc.rd as usize] = cpu.state.pc;
+                cpu.state.pc = enc.imm; 
+            },
+            repr: |enc| format!("jal {}", enc.imm), //probably should be better
+        });
+        m.insert(Opcode::JMP as u8, Operation {
+            execute: |enc, cpu| {
+                let jmp_code: u8 = ((enc.rs1 << 3) as u8) +  enc.rd as u8;
+                let cpu_flags: Flags = convert_to_bitflags(cpu.state.flags);
+
+                match jmp_code {
+                    0x0 => cpu.state.pc = enc.imm,
+                    0x1 => if cpu_flags & Flags::C == Flags::C { cpu.state.pc = enc.imm},
+                    0x2 => if cpu_flags & Flags::Z == Flags::Z { cpu.state.pc = enc.imm},
+                    0x3 => if cpu_flags & Flags::N == Flags::N { cpu.state.pc = enc.imm},
+                    0x4 => if cpu_flags & (Flags::N | Flags::Z) != (Flags::N | Flags::Z) { cpu.state.pc = enc.imm },
+                    0x5 => if cpu_flags & (Flags::N | Flags::Z) == (Flags::N | Flags::Z) { cpu.state.pc = enc.imm },
+                    0x6 => if cpu_flags & Flags::N != Flags::N { cpu.state.pc = enc.imm},
+                    0x7 => if cpu_flags & Flags::Z != Flags::Z { cpu.state.pc = enc.imm},
+                    0x8 => if cpu_flags & Flags::O == Flags::O { cpu.state.pc = enc.imm},
+                    0x9 => if cpu_flags & Flags::P == Flags::P { cpu.state.pc = enc.imm},
+                    0xA => if cpu_flags & (Flags::C | Flags::Z) != (Flags::C | Flags::Z) { cpu.state.pc = enc.imm },
+                    0xB => if cpu_flags & Flags::C != Flags::C { cpu.state.pc = enc.imm},
+                    0xC => if cpu_flags & (Flags::C | Flags::Z) == (Flags::C | Flags::Z) { cpu.state.pc = enc.imm },
+                    _ => println!("jmp jump_code error! invalid instruction."),
+                }
+
+
+
+
+            },
+            repr: |enc| { format!("jumped to {}", enc.imm) }, //looks ugly... TODO: do it better
```

Comment:
Use similar `match` statement for generating opcode string (jmp, jgt...) and then append `enc.imm` to it